From 07a30f561661efae1691ff45d10ec6014b395b58 Mon Sep 17 00:00:00 2001
From: Max Smolens <msmolens@gmail.com>
Date: Tue, 6 Oct 2020 17:58:52 -0400
Subject: [PATCH] Fixed #31850 -- Fixed
 BasicExtractorTests.test_extraction_warning with xgettext 0.21+.

"format string with unnamed arguments cannot be properly localized"
warning is not raised in xgettext 0.21+.

This patch uses a message that causes an xgettext warning regardless of
the version.
---
 AUTHORS                         | 1 +
 tests/i18n/commands/code.sample | 4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/AUTHORS b/AUTHORS
index b10852f407a4..9253740857de 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -622,6 +622,7 @@ answer newbie questions, and generally made Django that much better:
     mattycakes@gmail.com
     Max Burstein <http://maxburstein.com>
     Max Derkachev <mderk@yandex.ru>
+    Max Smolens <msmolens@gmail.com>
     Maxime Lorant <maxime.lorant@gmail.com>
     Maxime Turcotte <maxocub@riseup.net>
     Maximilian Merz <django@mxmerz.de>
diff --git a/tests/i18n/commands/code.sample b/tests/i18n/commands/code.sample
index a5f1520ecba5..2c305a3a1dcf 100644
--- a/tests/i18n/commands/code.sample
+++ b/tests/i18n/commands/code.sample
@@ -1,4 +1,4 @@
 from django.utils.translation import gettext
 
-# This will generate an xgettext warning
-my_string = gettext("This string contain two placeholders: %s and %s" % ('a', 'b'))
+# This will generate an xgettext "Empty msgid" warning.
+my_string = gettext('')
